HLTrigger/special/plugins/HLTMuonRecHitClusterFilter.cc

```diff
@@ -89,7 +101,11 @@ bool HLTMuonRecHitClusterFilter::filter(edm::StreamID, edm::Event& iEvent, const
   auto const& rechitClusters = iEvent.get(cluster_token_);
 
   for (auto const& cluster : rechitClusters) {
-    if ((cluster.size() >= min_Size_) && ((cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_) &&
+    if (((cluster.size() >= min_Size_) || ((cluster.size() - cluster.nMB1()) >= min_SizeMinusMB1_) ||
+         (std::abs(cluster.eta()) <= 1.9 && cluster.nStation() > 1 && cluster.size() >= min_SizeEtaLT1p9NstationGT1_) ||
+         (std::abs(cluster.eta()) <= 1.9 && cluster.nStation() <= 1 && cluster.size() >= min_SizeEtaLT1p9NstationEq1_) ||
+         (std::abs(cluster.eta()) > 1.9 && cluster.nStation() > 1 && cluster.size() >= min_SizeEtaGT1p9NstationGT1_) ||
+         (std::abs(cluster.eta()) > 1.9 && cluster.nStation() <= 1 && cluster.size() >= min_SizeEtaGT1p9NstationEq1_)) &&
```

Comment:
1. The complication arises because the input collection `MuonRecHitClusterCollection` can be constructed with DT rechits and CSC rechits. The logic could look nicer if we choose to have a string parameter of e.g. `clusterType=["DT","CSC"]`, then we do not need to select on `nMB1` for CSC clusters, and `time/timeSpread` for DT
 clusters, which do not have timing information. For longer term, we could have a mixture cluster type constructed with DT&CSC rechits, so `clusterType` would be better than `isCSCcluster`.
  
2. `1.9` is the eta boundary between the first ring and 2nd ring of the muon chambers, that should qualify for magic numbers? `Nstation` could be configurable.